libbpf-rs/src/program.rs

```diff
@@ -364,6 +366,34 @@ pub enum ProgramAttachType {
     Unknown = u32::MAX,
 }
 
+/// The input a program accepts.
+///
+/// This type is mostly used in conjunction with the [`Program::invoke`]
+/// facility.
+#[derive(Debug, Default)]
+pub struct Input<'dat> {
+    /// The input context to provide.
+    pub context: Option<&'dat [u8]>,
+    /// Additional data to provide to the program.
+    pub data: Option<&'dat [u8]>,
+    /// The struct is non-exhaustive and open to extension.
```

Comment:
I think at least `cpu` and `flags` are pretty straightforward to add and useful to have from the get go